server/updater.js

```diff
@@ -0,0 +1,182 @@
+const path = require('path');
+const axios = require('axios');
+const fs = require('fs');
+
+const {app, dialog} = require('electron');
+const {pipeline} = require('stream/promises');
+
+const Config = require('./config');
+const log = require('./logger').GetLogger('Updater');
+
+const AppDataPath = app.getPath('userData');
+const UpdaterPath = path.join(AppDataPath, 'CVRUpdater');
+
+const binaryContentType = 'application/octet-stream';
+const windowsPlatform = 'win32';
+const latestRelease = 'https://api.github.com/repos/AstroDogeDX/CVRX/releases/latest';
+
+const currentVersion = `v${app.getVersion()}`;
+
+const checkUpdatesIntervalMinutes = 15;
+const activelyCheckRateLimit = 2 * 60 * 1000;
+
+let ignoredForNow = false;
+let dialogOpened = false;
+let activelyCheckForUpdatesTimesout = null;
+
+exports.Setup = async (mainWindow) => {
+
+    // Ensure we have the directory created
+    await fs.promises.mkdir(UpdaterPath, {recursive: true});
+
+    // Clear installer filers
+    for (const fileName of await fs.promises.readdir(UpdaterPath)) {
+        if (!fileName.endsWith('.exe')) continue;
+        const filePath = path.join(UpdaterPath, fileName);
+        log.info(`[ClearInstallerFiles] Clearing previous update file: ${filePath}`);
+        await fs.promises.unlink(filePath);
+    }
+
+    setInterval(exports.CheckLatestRelease, checkUpdatesIntervalMinutes * 60 * 1000, mainWindow);
+};
+
+exports.CheckLatestRelease = async (mainWindow, bypassIgnores = false) => {
+
+    if (bypassIgnores) {
+        if (activelyCheckForUpdatesTimesout && activelyCheckForUpdatesTimesout > Date.now()) {
+            const msg = `Manually checked for updates too recently.
+            You can try again in: ${(activelyCheckForUpdatesTimesout - Date.now()) / 1000} seconds.`;
+            log.info(`[CheckLatestRelease] ${msg}`);
+            return { hasUpdates: null, msg: msg };
+        }
+        activelyCheckForUpdatesTimesout = Date.now() + activelyCheckRateLimit;
+    }
+
+    if (dialogOpened) {
+        const msg = 'Skipping because the previous dialog is still open';
+        log.info(`[CheckLatestRelease] ${msg}`);
+        return { hasUpdates: null, msg: msg };
+    }
+
+    if (!bypassIgnores && ignoredForNow) {
+        const msg = 'Skipping check since user requested to ignore...';
+        log.info(`[CheckLatestRelease] ${msg}`);
+        return { hasUpdates: null, msg: msg };
+    }
+
+    try {
+
+        log.info('[CheckLatestRelease] Checking for updates...');
+
+        const {data} = await axios.get(latestRelease);
+        const tagName = data.tag_name;
+
+        if (tagName !== currentVersion) {
+            log.warn(`[CheckLatestRelease] There is a new version available (${tagName}), installed version: ${currentVersion}`);
+        }
+        else {
+            const msg = `You have the current latest version: ${tagName}!`;
+            log.info(`[CheckLatestRelease] ${msg}`);
+            return { hasUpdates: false, msg: msg };
```

Comment:
If you want to test, comment this return. It will update you to the current 0.0.2 that's on the releases.